Source/WebKit/UIProcess/Automation/WebDriverBidiProcessor.cpp

```diff
@@ -136,11 +306,84 @@ Inspector::Protocol::ErrorStringOr<void> WebDriverBidiProcessor::close()
     return { };
 }
 
+// MARK: Log domain.
+
 void WebDriverBidiProcessor::logEntryAdded(const String& level, const String& source, const String& message, double timestamp, const String& type, const String& method)
 {
     m_logDomainNotifier->entryAdded(level, source, message, timestamp, type, method);
 }
 
+// MARK: Inspector::BidiScriptDispatcherHandler methods.
+
+void WebDriverBidiProcessor::callFunction(const String& functionDeclaration, bool awaitPromise, Ref<JSON::Object>&& target, RefPtr<JSON::Array>&& arguments, std::optional<Protocol::BidiScript::ResultOwnership>&&, RefPtr<JSON::Object>&& optionalSerializationOptions, RefPtr<JSON::Object>&& optionalThis, std::optional<bool>&& optionalUserActivation, CommandCallbackOf<Protocol::BidiScript::EvaluateResultType, String, RefPtr<Protocol::BidiScript::RemoteValue>, RefPtr<Protocol::BidiScript::ExceptionDetails>>&& callback)
+{
+    // FIXME: handle `awaitPromise` option.
+    // FIXME: handle `resultOwnership` option.
+    // FIXME: handle `serializationOptions` option.
+    // FIXME: handle custom `this` option.
+    // FIXME: handle `userActivation` option.
+
+    RefPtr session = m_session.get();
+    if (!session)
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(InternalError);
+
+    // FIXME: handle non-BrowsingContext obtained from `Target`.
+    std::optional<BrowsingContext> browsingContext = target->getString("context"_s);
+    if (!browsingContext || !session->webPageProxyForHandle(*browsingContext))
+        ASYNC_FAIL_WITH_PREDEFINED_ERROR(WindowNotFound);
+
+    Ref<JSON::Array> outArguments = JSON::Array::create();
+    if (arguments) {
+        for (auto argument : *arguments)
+            outArguments->pushValue(argument.copyRef());
+    }
```

Comment:
why do we need to create a copy of `arguments`?

Reply:
`RefPtr<JSON::Object>` is an optional argument, but the call to `WebAutomationSession::evaluateJavascriptFunction` wants a non-null, possibly empty, `JSON::Object`.

Reply:
couldn't we just do something like this?
```cpp
arguments ? arguments.releaseNonNull() : JSON::Object::create()
```